src/metabase/notification/seed.clj

```diff
@@ -0,0 +1,113 @@
+                                              :channel_type "channel/email"
+                                              :details      {:type           "email/handlebars-resource"
+                                                             :subject        "Your Slack connection stopped working"
+                                                             :path           "metabase/email/slack_token_error.hbs"
+                                                             :recipient-type :cc}
+                                              :created_at   :%now
+                                              :updated_at   :%now}
+                            :recipients      [{:type    :notification-recipient/template
+                                               :details {:pattern     "{{context.admin_email}}"
+                                                         :is_optional true}}
+                                              {:type                 :notification-recipient/group
+                                               :permissions_group_id (:id (perms-group/admin))}]}]}]))
+
+(def ^:private notification-related-models
+  [:model/Notification
+   :model/NotificationSubscription
+   :model/NotificationHandler
+   :model/NotificationRecipient
+   :model/ChannelTemplate])
+
+(defn- truncate-table!
+  [table-name]
+  (t2/query {:truncate [table-name :restart :identity :cascade]}))
+
+(defn- truncate-notification-related-tables!
+  []
+  (doseq [model notification-related-models
+          :let [table (t2/table-name model)]]
+    (log/infof "Truncating table %s" table)
+    (truncate-table! table)))
+
+(defn- create-notification!
+  [{:keys [notification subscriptions handlers]}]
+  (let [handlers (for [handler handlers]
+                   (if-let [template (:template handler)]
+                     (-> handler
+                         (dissoc :template)
+                         (assoc :template_id (t2/insert-returning-pk!
+                                              :model/ChannelTemplate
+                                              template)))
+                     handler))]
+    (models.notification/create-notification! notification subscriptions handlers)))
+
+(defn- seed-notification!
+  "Seed default notifications"
+  []
+  (log/info "Seeding default notifications")
+  (doseq [row @default-notifications]
+    (create-notification! row))
+  (log/infof "Seeded %d notifications" (count @default-notifications)))
+
+(defn truncate-then-seed-notification!
+  "Truncate all notifications related tables then seed it with the default notifications.
+  Runs on every deployment by metabase.task.seed_notification.
+
+  We do this instead of migrations because we want to be able to modify the default notifications
+  without having to write migrations.
+
+  This is possible because as of now users can't modify notifications or channel template."
```

Comment:
Uhm right but why truncate though? I thought the idea is the following: fetch what's in the db, check if it changed, and insert what's not in there yet, plus update all unchanged (possibly also remove unchanged but unknown to us notification, like if we remove some notification).

Reply:
yea but I figured we don't have to do it now as we haven't let users modify it.

this is way simpler, I think.

Reply:
Well it's simpler but we'll have to modify this code again :)

Reply:
I think it's fine? Since in order to do that now, I think we'll need a way to identify notifications, probably by creating a `notification.name` and making it unique.

Reply:
Also I think it's better to do this once we work on "reset to factory" template so we can think the whole thing through https://www.notion.so/metabase/Customizable-notification-templates-11f69354c9018055a33cd21472a7fb4c?pvs=4